apps/hash-graph/lib/graph/src/store/postgres/query/table.rs

```diff
@@ -479,97 +474,134 @@ pub enum Relation {
     IncomingLink,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum ForeignKeyReference {
+    Single {
+        on: Column<'static>,
+        join: Column<'static>,
+    },
+    Double {
+        on: (Column<'static>, Column<'static>),
+        join: (Column<'static>, Column<'static>),
+    },
```

Comment:
Any reason for this being split up like this instead of just defining the column names in something like a const array or equivalent? Could probably simplify the `from_foreign_key` function if we change the representation

Reply:
I changed the definition of `Double` to be `[Colum<'static>; 2]` instead, which makes it more readable. I was not able to make the number of columns static as it's not known at compile time as this comes from user input enumerations. I used this approach over just a `Vec<_>` to have better control over the size of the join statement and avoid unneeded allocations, as otherwise, we'd need a vector for each join.